ngx-graph-frontend/src/app/sidebar/sidebar.component.ts

```diff
@@ -46,7 +46,30 @@ export class SidebarComponent implements OnInit {
         alert("Graph " + name + " already exists. To modify it, please select it from the dropdown.")
         return;
       }
-    } 
+    }
     this.graphService.saveGraph(name);
   }
+
+  onFileChanged(event) {
+    if (event.target.files && event.target.files[0]) {
+      const fileReader: FileReader = new FileReader();
+      fileReader.readAsText(event.target.files[0], "UTF-8");
+      fileReader.onload = () => {
+        if (typeof fileReader.result === "string") {
+          const graphJson: any = JSON.parse(fileReader.result);
+          for (let node of graphJson.graph.nodes) {
```

Comment:
This will mess up current graph.
addEdge / addNode adds edges/nodes to currentGraph.

You'll have to add a new graph to the list of graphs and modify it's contents.